src/posix/platform/radio_spinel.cpp

```diff
@@ -1337,6 +1337,37 @@ otError RadioSpinel::PlatDiagProcess(const char *aString, char *aOutput, size_t
 }
 #endif
 
+uint32_t RadioSpinel::GetRadioChannelMask(spinel_prop_key_t aKey)
+{
+    uint8_t        maskBuffer[kChannelMaskBufferSize];
+    otError        error       = OT_ERROR_NONE;
+    uint32_t       channelMask = 0;
+    const uint8_t *maskData    = maskBuffer;
+    spinel_size_t  maskLength  = sizeof(maskBuffer);
+
+    VerifyOrExit(aKey == SPINEL_PROP_PHY_CHAN_SUPPORTED || aKey == SPINEL_PROP_PHY_CHAN_PREFERRED,
+                 error = OT_ERROR_INVALID_ARGS);
+    SuccessOrExit(error = Get(aKey, SPINEL_DATATYPE_DATA_S, maskBuffer, &maskLength));
+
+    while (maskLength > 0)
+    {
+        uint8_t        channel;
+        spinel_ssize_t unpacked;
+
+        unpacked = spinel_datatype_unpack(maskData, maskLength, SPINEL_DATATYPE_UINT8_S, &channel);
+        VerifyOrExit(unpacked > 0, error = OT_ERROR_FAILED);
+        VerifyOrExit(channel < kChannelMaskBufferSize, error = OT_ERROR_INVALID_ARGS);
```

Comment:
#define as `kMaxChannel`? And maybe we should just drop unsupported channel or return parse error.

Reply:
Here the kChannelMaskBufferSize (32) is different from kMaxChannel (normally 26). Updated to OT_ERROR_PARSE.